### PR TITLE
chore: make rest api node port service configurable

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
Signed-off-by: Abhinandan Purkait <purkaitabhinandan@gmail.com>

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -178,6 +178,17 @@ apis:
         cpu: "50m"
         # -- Memory requests for rest
         memory: "32Mi"
+    # Rest service parameters defines how the rest service is exposed
+    service:
+      # K8s service type
+      type: ClusterIP
+
+      # Ports from where rest endpoints are accessible from outside cluster, only valid if type is NodePort
+      nodePorts:
```
Comment (blaisedias) [inline chart/values.yaml:187]: Should these lines be commented out by default?

Comment (avishnu) [reply chart/values.yaml:187]: Yes IMO.

File: chart/values.yaml [hunk 2]
```diff
@@ -178,6 +178,17 @@ apis:
         cpu: "50m"
         # -- Memory requests for rest
         memory: "32Mi"
+    # Rest service parameters defines how the rest service is exposed
+    service:
+      # K8s service type
+      type: ClusterIP
+
```
Comment (avishnu) [inline chart/values.yaml:185]: Blank line can be removed.

File: chart/values.yaml [hunk 3]
```diff
@@ -178,6 +178,17 @@ apis:
         cpu: "50m"
         # -- Memory requests for rest
         memory: "32Mi"
+    # Rest service parameters defines how the rest service is exposed
```
Comment (avishnu) [inline chart/values.yaml:181]: typo: define

File: chart/values.yaml [hunk 4]
```diff
@@ -178,6 +178,17 @@ apis:
...
+    service:
+      # K8s service type
+      type: ClusterIP
+
+      # Ports from where rest endpoints are accessible from outside cluster, only valid if type is NodePort
```
Comment (avishnu) [inline chart/values.yaml:186]: outside the cluster

**Comment by avishnu** [comment]
Thanks @Abhinandan-Purkait . Do we need similar changes for Loki and etcd?

File: chart/values.yaml (hunk 1 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml:187]: No, these values are also configurable. It will not work if we comment them.

File: chart/values.yaml (hunk 3 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml:181]: Done.

**Comment by Abhinandan-Purkait** [comment]
> Thanks @Abhinandan-Purkait . Do we need similar changes for Loki and etcd?

For loki and etcd its already there.

File: chart/values.yaml (hunk 4 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml:186]: Done.

File: chart/values.yaml (hunk 1 above)
Comment (avishnu) [reply chart/values.yaml:186]: Seeing similar configuration for Loki and etcd as well. Even though, by default, service type is clusterIp, there is a nodePort value as well.

Comment (Abhinandan-Purkait) [reply chart/values.yaml:186]: No, we have an if conditional in the template that would add/remove this based on the types.
This in similar to what other chart providers follow.:- ref etcd https://github.com/bitnami/charts/blob/610f860fd8197f6f1dc1906511b56a96df42dac9/bitnami/etcd/values.yaml#L517

Comment (avishnu) [reply chart/values.yaml:186]: Having said that, we can still comment out these lines by default, specifying in the comment that "Uncomment below lines if serviceType is nodePort" - @Abhinandan-Purkait

Comment (Abhinandan-Purkait) [reply chart/values.yaml:186]: Okay. Shall I comment them for `etcd` and `loki-stack` as well, so that its similar throughout.

Comment (tiagolobocastro) [reply chart/values.yaml:186]: I don't think we should comment them out. They're there as the default in case node port is used.

Comment (avishnu) [reply chart/values.yaml:186]: As discussed, looks good, no need to comment.

**Review by avishnu — APPROVED** [review]
lgtm

File: chart/values.yaml [hunk 5]
```diff
@@ -178,6 +178,16 @@ apis:
...
         # -- Memory requests for rest
         memory: "32Mi"
+    # Rest service parameters define how the rest service is exposed
+    service:
+      # -- Rest K8s service type
+      type: ClusterIP
+      # Ports from where rest endpoints are accessible from outside the cluster, only valid if type is NodePort
+      nodePorts:
+        # NodePort associated with http port
+        httpPort: 30011
```
Comment (tiagolobocastro) [inline chart/values.yaml:188]:
```suggestion
        http: 30011
```

File: chart/values.yaml [hunk 6]
```diff
@@ -178,6 +178,16 @@ apis:
...
+        # NodePort associated with https port
+        httpsPort: 30010
```
Comment (tiagolobocastro) [inline chart/values.yaml:190]:
```suggestion
        https: 30010
```

**Review by tiagolobocastro — APPROVED** [review]

File: chart/values.yaml (hunk 5 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml]: Done.

File: chart/values.yaml (hunk 6 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors try

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**